cucumber-junit/src/main/java/io/cucumber/junit/NoUuidGenerator.java

```diff
@@ -0,0 +1,20 @@
+package io.cucumber.junit;
+
+import io.cucumber.core.eventbus.UuidGenerator;
+
+import java.util.UUID;
+
+/**
+ * This UUID generator does nothing. It is solely needed for marking purposes.
+ */
+final class NoUuidGenerator implements UuidGenerator {
+
+    private NoUuidGenerator() {
+        // No need for instantiation
+    }
+
+    @Override
+    public UUID get() {
+        return null;
```

Comment:
Note to self: Write an issue with the considerations of using a null-safe API, returning null only on methods annotated with nullable. Specifically investigate automated checks.